opentelemetry-sdk/src/logs/record.rs

```diff
@@ -118,7 +118,56 @@ impl opentelemetry::logs::LogRecord for LogRecord {
 }
 
 impl LogRecord {
+    /// Returns the event name
+    #[inline]
+    pub fn event_name(&self) -> Option<&'static str> {
+        self.event_name
+    }
+
+    /// Returns the target
+    #[inline]
+    pub fn target(&self) -> Option<&Cow<'static, str>> {
+        self.target.as_ref()
+    }
```

Comment:
Returning &str would simplify the API, but want to preserve the ability for consumers to optimize based on whether the target is a static string.

Reply:
I think we could favor simplicity of the API here and use `&str` as the return type. The current exporters that we have cannot utilize the `static` str reference to their benefit anyway.

Only exporters that would benefit from `Cow<'static, str>` would be the ones that hold these values in memory for some reason as they could then avoid the cost to clone a static str.

Reply:
Not a strong opinion here as it's mainly used by exporters and not end users. We can expect exporter authors to work with `Cow<'static,str>`.

Reply:
> Not a strong opinion here as it's mainly used by exporters and not end users. We can expect exporter authors to work with Cow<'static,str>.

Yes, that's what I thought - keeping it for the custom export authors if they want to reap the benefit of static target.